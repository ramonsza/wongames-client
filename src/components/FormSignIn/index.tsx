import Link from 'next/link'
import { EnvelopeSimple as MailIcon } from '@phosphor-icons/react'
import { LockKey as LockIcon } from '@phosphor-icons/react'
import TextField from '@/components/TextField'
import { FormWrapper, FormLink } from '@/components/Form/'
import * as S from './styles'
import Button from '@/components/Button'

const FormSignIn = () => (
  <FormWrapper>
    <form>
      <TextField
        name="email"
        placeholder="Email"
        type="email"
        icon={<MailIcon size={22} />}
      />
      <TextField
        name="password"
        placeholder="Password"
        type="password"
        icon={<LockIcon size={22} />}
      />
      <S.ForgotPassword href="#">Forgot your password?</S.ForgotPassword>
      <Button $size="large" $fullWidth>
        Sign in now
      </Button>

      <FormLink>
        Don&apos;t have an account <Link href="/sign-up">Sign up</Link>
      </FormLink>
    </form>
  </FormWrapper>
)

export default FormSignIn
