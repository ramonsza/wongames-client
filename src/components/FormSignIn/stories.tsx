import { Meta, StoryObj } from '@storybook/react'
import FormSignIn from '.'

export default {
  title: 'Form/FormSignIn',
  component: FormSignIn
} as Meta

export const Default: StoryObj = {
  render: () => (
    <div style={{ width: '300px', margin: 'auto' }}>
      <FormSignIn />
    </div>
  )
}
