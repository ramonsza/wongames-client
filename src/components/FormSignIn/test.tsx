import { screen } from '@testing-library/react'

import FormSignIn from '.'
import { renderWithTheme } from '@/utils/tests/helpers'

describe('<FormSignIn />', () => {
  it('should render the form', () => {
    const { container } = renderWithTheme(<FormSignIn />)

    expect(screen.getByPlaceholderText(/email/i)).toHaveAttribute(
      'type',
      'email'
    )
    expect(screen.getByPlaceholderText(/password/i)).toHaveAttribute(
      'type',
      'password'
    )

    expect(
      screen.getByRole('button', { name: /Sign in now/i })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render the forgot password link', () => {
    renderWithTheme(<FormSignIn />)

    expect(
      screen.getByRole('link', { name: /Forgot your password\?/i })
    ).toBeInTheDocument()
  })

  it('should render the text and link to sign up', () => {
    renderWithTheme(<FormSignIn />)

    expect(screen.getByRole('link', { name: /sign up/i })).toBeInTheDocument()
    expect(screen.getByText(/don't have an account/i)).toBeInTheDocument()
  })
})
