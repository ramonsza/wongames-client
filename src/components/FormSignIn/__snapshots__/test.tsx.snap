// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render the form 1`] = `
.c2 {
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c3 {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 2.2rem;
  color: #8F8F8F;
  order: 0;
}

.c3>svg {
  width: 100%;
}

.c4 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c7 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c7:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c6 {
  margin: 3.2rem auto 1.6rem;
}

.c8 {
  font-size: 1.4rem;
  color: #030517;
  text-align: center;
}

.c8 a {
  color: #3CD3C1;
  text-decoration: none;
  border-bottom: 0.1rem solid #3CD3C1;
  transition: color,border,0.1s ease-in-out;
}

.c8 a:hover {
  color: #29b3a3;
  border-bottom: 0.1rem solid #29b3a3;
}

.c5 {
  display: block;
  font-size: 1.4rem;
  color: #030517;
  text-decoration: none;
  text-align: right;
}

.c5:hover {
  color: #0f1971;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                fill="currentColor"
                height="22"
                viewBox="0 0 256 256"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224,48H32a8,8,0,0,0-8,8V192a16,16,0,0,0,16,16H216a16,16,0,0,0,16-16V56A8,8,0,0,0,224,48ZM203.43,64,128,133.15,52.57,64ZM216,192H40V74.19l82.59,75.71a8,8,0,0,0,10.82,0L216,74.19V192Z"
                />
              </svg>
            </div>
            <input
              class="c4"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                fill="currentColor"
                height="22"
                viewBox="0 0 256 256"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,112a28,28,0,0,0-8,54.83V184a8,8,0,0,0,16,0V166.83A28,28,0,0,0,128,112Zm0,40a12,12,0,1,1,12-12A12,12,0,0,1,128,152Zm80-72H176V56a48,48,0,0,0-96,0V80H48A16,16,0,0,0,32,96V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V96A16,16,0,0,0,208,80ZM96,56a32,32,0,0,1,64,0V80H96ZM208,208H48V96H208V208Z"
                />
              </svg>
            </div>
            <input
              class="c4"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <a
          class="c5"
          href="#"
        >
          Forgot your password?
        </a>
        <button
          class="c6 c7"
        >
          <span>
            Sign in now
          </span>
        </button>
        <div
          class="c8"
        >
          Don't have an account 
          <a
            href="/sign-up"
          >
            Sign up
          </a>
        </div>
      </form>
    </div>
  </div>
</body>
`;
